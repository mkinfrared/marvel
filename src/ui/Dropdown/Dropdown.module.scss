@import "styles/colors";

.Dropdown {
  border-radius: 5px;
  box-shadow: 0 0 4px rgba(0, 0, 0, 0.2);
  overflow: hidden;

  .list {
    display: flex;
    flex-direction: column;

    > * {
      align-items: center;
      background-color: #FFFFFF;
      border-bottom: 1px solid $secondary-main;
      cursor: pointer;
      display: flex;
      font-family: Roboto, sans-serif;
      list-style-type: none;
      padding: 10px;
      text-align: left;

      > p {
        overflow: hidden;
        text-overflow: ellipsis;
        white-space: nowrap;
        width: 100%;
      }

      &:last-child {
        border-bottom: none;
      }
    }
  }

  .loading {
    background-color: #FFFFFF;
    display: flex;
    justify-content: center;
    padding: 20px;
  }
}
