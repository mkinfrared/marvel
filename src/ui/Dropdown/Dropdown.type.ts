export interface DropdownProps {
  children?: React.ReactNode;
  className?: string;
  loading?: boolean;
}
