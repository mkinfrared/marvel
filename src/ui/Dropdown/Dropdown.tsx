import React from "react";

import Spinner from "ui/Spinner";
import classes from "utils/classes";

import css from "./Dropdown.module.scss";
import { DropdownProps } from "./Dropdown.type";

const Dropdown = ({ children, className, loading = false }: DropdownProps) => {
  const getContent = () => {
    if (loading) {
      return (
        <div className={css.loading}>
          <Spinner size={2} />
        </div>
      );
    }

    return <div className={css.list}>{children}</div>;
  };

  return <div className={classes(css.Dropdown, className)}>{getContent()}</div>;
};

export default React.memo(Dropdown);
