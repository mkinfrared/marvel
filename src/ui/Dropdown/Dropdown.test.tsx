import { render } from "@testing-library/react";
import React from "react";

import Dropdown from "./Dropdown";

describe("<Dropdown />", () => {
  const Component = <Dropdown />;

  it("should be defined", () => {
    const { container } = render(Component);

    expect(container).toBeDefined();
  });

  it("should match the snapshot", () => {
    const { container } = render(Component);

    expect(container).toMatchSnapshot();
  });

  it("should render list block with children", () => {
    const children = (
      <>
        <li>1</li>
        <li>2</li>
      </>
    );

    const Comp = <Dropdown>{children}</Dropdown>;
    const { container } = render(Comp);
    const list = container.querySelector(".list");
    const spinner = container.querySelector(".Spinner");

    expect(list).toBeInTheDocument();

    expect(list!.children).toHaveLength(2);

    expect(spinner).not.toBeInTheDocument();
  });

  it("should render a spinner", () => {
    const children = (
      <>
        <li>1</li>
        <li>2</li>
      </>
    );

    const Comp = <Dropdown loading>{children}</Dropdown>;
    const { container } = render(Comp);
    const list = container.querySelector(".list");
    const spinner = container.querySelector(".Spinner");

    expect(list).not.toBeInTheDocument();

    expect(spinner).toBeInTheDocument();
  });
});
