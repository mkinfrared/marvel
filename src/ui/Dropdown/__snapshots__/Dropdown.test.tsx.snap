// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dropdown /> should match the snapshot 1`] = `
<div>
  <div
    class="Dropdown"
  >
    <div
      class="list"
    />
  </div>
</div>
`;
